bitcoin/script/miniscript.h

```diff
@@ -1408,7 +1408,11 @@ inline NodeRef<Key> DecodeScript(I& in, I last, const Ctx& ctx) {
         }
     }
     if (constructed.size() != 1) return {};
-    return constructed.front();
+    const NodeRef<Key> tl_node = constructed.front();
```

Comment:
Put a `std::move` around `constructed.front();` to avoid an `std::shared_ptr` copy (it's a tiny cost, but much more than just a pointer copy due to atomic refcount updating).